packages/snowpack/src/commands/build.ts

```diff
@@ -147,55 +146,63 @@ export async function command(commandOptions: CommandOptions) {
 
   for (const [dirDisk, dirDest, allFiles] of includeFileSets) {
     for (const locOnDisk of allFiles) {
-      const {baseExt: fileExt} = getExt(locOnDisk);
-      let outLoc = locOnDisk.replace(dirDisk, dirDest);
-      let builtLocOnDisk = locOnDisk;
-      const extToReplace = config._extensionMap[fileExt] || srcFileExtensionMapping[fileExt];
-      if (extToReplace) {
-        outLoc = replaceExt(outLoc, extToReplace);
-        builtLocOnDisk = replaceExt(builtLocOnDisk, extToReplace);
-      }
+      const replaceSrcExt = new RegExp(path.extname(locOnDisk) + '$');
       const builtFileOutput = await buildFile(locOnDisk, {
         plugins: config.plugins,
         messageBus,
         isDev: false,
         isHmrEnabled: false,
       });
       allBuiltFromFiles.add(locOnDisk);
-      const {baseExt, expandedExt} = getExt(outLoc);
-      let contents =
-        builtFileOutput[
-          config._extensionMap[fileExt] || srcFileExtensionMapping[fileExt] || fileExt
-        ];
-      if (!contents) {
-        continue;
-      }
-      const cssOutPath = outLoc.replace(/.js$/, '.css');
-      mkdirp.sync(path.dirname(outLoc));
-      switch (baseExt) {
-        case '.js': {
-          if (builtFileOutput['.css']) {
-            await fs.mkdir(path.dirname(cssOutPath), {recursive: true});
-            await fs.writeFile(cssOutPath, builtFileOutput['.css'], 'utf-8');
-            contents = `import './${path.basename(cssOutPath)}';\n` + contents;
-          }
-          contents = wrapImportMeta({code: contents, env: true, isDev: false, hmr: false, config});
-          allFilesToResolveImports[outLoc] = {baseExt, expandedExt, contents, locOnDisk};
-          break;
+
+      for (const [fileExt, buildResult] of Object.entries(builtFileOutput)) {
```

Comment:
⚠️ This is actually a pretty big change. I’m not sure if we were doing this intentionally or not, but it seemed here we were going against our own design.

When a plugin has multiple outputs (e.g. for Svelte this PR: `.js`, `.js.map`, `.css`, `.css.map`), we were only doing **one pass** for all of those! So at first, adding a `.js.map` to the output didn’t generate anything at all 😅. The reason we didn’t notice was because we were manually writing `.css` files, and we were only testing the `.js` + `.css` output case before.

This changes this logic to now write all keys of the file, which I think is better. But the question is: will this likely break any legacy plugins? I think not, but wanted to flag it.

Reply:
ah good catch! Yup, I bet we just grabbed our old logic here without updating it to the new build map format. This should be good to merge, I can't think of what it could be breaking that isn't already broken